best-practices/ml-platform/terraform/modules/node-pools/nodepools.tf

```diff
@@ -90,3 +91,31 @@ resource "google_container_node_pool" "node-pool" {
     update = "20m"
   }
 }
+
+# Create dedicated service account for node pool
+resource "google_service_account" "nodepool_sa" {
+  project = var.project_id
+  account_id = "${var.node_pool_name}-sa"
```

Comment:
Can you prefix the account_id with `vm-`

https://cloud.google.com/iam/docs/best-practices-service-accounts#naming-convention

Also the `-sa` is redundant.